File: Java/E-montazysta/src/main/java/com/emontazysta/model/Notification.java
package com.emontazysta.model;

import com.emontazysta.enums.NotificationType;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.*;
import org.hibernate.annotations.SQLDelete;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@SQLDelete(sql = "UPDATE notification SET deleted = true WHERE id=?")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Notification {

    public Notification(Long id, String content, LocalDateTime createdAt, LocalDateTime readAt, AppUser createdBy,
                        AppUser notifiedEmployee, OrderStage orderStage) {
        this.id = id;
        this.content = content;
        this.createdAt = createdAt;
        this.readAt = readAt;
        this.createdBy = createdBy;
        this.notifiedEmployee = notifiedEmployee;
        this.orderStage = orderStage;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private NotificationType notificationType;

    private String content;

    private String subContent;

    private LocalDateTime createdAt;

    private LocalDateTime readAt;

    private boolean deleted = Boolean.FALSE;

    @ManyToOne
    private AppUser createdBy;

    @ManyToOne
    private AppUser notifiedEmployee;

    @ManyToOne
    private OrderStage orderStage;

    @ManyToOne
    private Orders order;

    @ManyToOne
    private ToolEvent toolEvent;

    @ManyToOne
    private ElementEvent elementEvent;

}

File: Android/app/src/main/java/com/example/e_montazysta/ui/notification/Notification.kt
package com.example.e_montazysta.ui.notification


import com.example.e_montazysta.data.model.NotificationType
import com.example.e_montazysta.data.model.User
import java.util.Date

data class Notification(
    val content: String,
    val createdAt: Date,
    val createdBy: User,
    val deleted: Boolean,
    val elementEventId: Int?,
    val id: Int,
    val notificationType: NotificationType,
    val notifiedEmployeeId: Int,
    val orderStageId: Int?,
    val orderId: Int?,
    val readAt: Date?,
    val subContent: String?,
    val toolEventId: Int?
) {
    fun mapToNotificationListItem(): NotificationListItem {
        return NotificationListItem(
            id,
            content,
            subContent,
            createdAt.toString(),
            createdBy.id,
            elementEventId,
            orderStageId,
            orderId,
            readAt.toString(),
            toolEventId
        )
    }
}